webapp/src/js/controllers/contacts-report.js

```diff
@@ -120,7 +115,7 @@ angular.module('inboxControllers').controller('ContactsReportCtrl',
 
       ctrl.setEnketoSavingStatus(true);
       resetFormError();
-      Enketo.save($state.params.formId, ctrl.form, geolocation)
+      Enketo.save($state.params.formId, ctrl.form, geoHandle)
```

Comment:
I think it might be nicer to have the Enketo service depend on the Geolocation service. It could initiate it in the `render` function and then check it in the `save` function, and cancel it in the `unload` function. That way we don't have to micromanage it from the controllers.

Reply:
Yeah I agree this would better. I had a quick look awhile ago and it's kind of wonky because it's stateless, so I'm not clear what the implications are in terms of how to keep hold of handles etc. I know we can just ignore the statelessness but that seems... bad, to me. This may end up being an extra credit assignment for someone else!

Reply:
OK the handle we pass around is an `EnketoForm` object. How disgusted with me would you be if I just attached the geohandle to that and hope that Enketo doesn't notice consequentially?

Reply:
I don't think the Enketo service is as stateless as it appears... we have `objUrls` which is an array of media blobs for showing in the form, and listeners on HTML elements in the DOM that modify things in the EnketoForm, and so on. These are all cleaned up in the `Enketo.unload` function which is expected to be called when the form is unloaded. Regardless of whether this is a good design or not, I think there's precedence for adding the geo state to the Enketo service also.

I'm not really comfortable with sticking it on the EnketoForm object...

It's also fine to leave it as extra credit for next time!